lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -473,25 +469,42 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
     assert(
         lhsIsSeqOp != rhsIsSeqOp &&
         "unexpected sequential operation on both sides; please open an issue");
-    Operation *seqOp = calyx::parentIsSeqCell(cmpIOp.getLhs())
-                           ? cmpIOp.getLhs().getDefiningOp()
-                           : cmpIOp.getRhs().getDefiningOp();
-    condReg = getState<ComponentLoweringState>().getSeqResReg(seqOp);
+    // If `cmpIOp`'s lhs/rhs operand is the result of a sequential operation,
+    // its result will be stored in a register.
+    resReg =
+        cast<calyx::RegisterOp>(lhsIsSeqOp ? cmpIOp.getLhs().getDefiningOp()
+                                           : cmpIOp.getRhs().getDefiningOp());
 
     auto groupOp = cast<calyx::GroupOp>(group);
     getState<ComponentLoweringState>().addBlockScheduleable(cmpIOp->getBlock(),
                                                             groupOp);
 
-    buildAssignmentsForRegisterWrite(
-        rewriter, groupOp, getState<ComponentLoweringState>().getComponentOp(),
-        resReg, condReg.getOut());
+    rewriter.setInsertionPointToEnd(groupOp.getBodyBlock());
+    auto loc = cmpIOp.getLoc();
+    assert(
+        (isa<calyx::EqLibOp, calyx::NeqLibOp, calyx::SleLibOp, calyx::SltLibOp,
+             calyx::LeLibOp, calyx::LtLibOp, calyx::GeLibOp, calyx::GtLibOp,
+             calyx::SgeLibOp, calyx::SgtLibOp>(calyxOp.getOperation())) &&
+        "Must be a Calyx comparison library operation.");
```

Comment:
Ah, sorry I didn't realize we couldn't just do something like `isa<calyx::Comparison>(op)`.

Reply:
yeah.. no it's fine :)